activerecord/lib/active_record/associations/builder/belongs_to.rb

```diff
@@ -34,8 +34,9 @@ def belongs_to_counter_cache_after_update(reflection)
           foreign_key  = reflection.foreign_key
           cache_column = reflection.counter_cache_column
 
-          if (@_after_replace_counter_called ||= false)
-            @_after_replace_counter_called = false
+          @_after_replace_counter_called ||= {}
+          if @_after_replace_counter_called.has_key?(foreign_key)
```

Comment:
I thought `has_key?`was deprecated in favor of `key?`? I prefer the latter anyway.